internal/manifests/collector/adapters/config_to_ports.go

```diff
@@ -24,17 +24,91 @@ import (
 	"github.com/mitchellh/mapstructure"
 	corev1 "k8s.io/api/core/v1"
 
-	"github.com/open-telemetry/opentelemetry-operator/internal/manifests/collector/parser"
+	exporterParser "github.com/open-telemetry/opentelemetry-operator/internal/manifests/collector/parser/exporter"
+	receiverParser "github.com/open-telemetry/opentelemetry-operator/internal/manifests/collector/parser/receiver"
 )
 
 var (
+	// ErrNoExporters indicates that there are no exporters in the configuration.
+	ErrNoExporters = errors.New("no exporters available as part of the configuration")
+
 	// ErrNoReceivers indicates that there are no receivers in the configuration.
 	ErrNoReceivers = errors.New("no receivers available as part of the configuration")
 
 	// ErrReceiversNotAMap indicates that the receivers property isn't a map of values.
 	ErrReceiversNotAMap = errors.New("receivers property in the configuration doesn't contain valid receivers")
+
+	// ErrExportersNotAMap indicates that the receivers property isn't a map of values.
+	ErrExportersNotAMap = errors.New("exporters property in the configuration doesn't contain valid exporters")
 )
 
+// ConfigToExporterPorts converts the incoming configuration object into a set of service ports required by the exporters.
+func ConfigToExporterPorts(logger logr.Logger, config map[interface{}]interface{}) ([]corev1.ServicePort, error) {
+	// now, we gather which ports we might need to open
+	// for that, we get all the receivers and check their `endpoint` properties,
```

Comment:
receivers? 

It seems the following lines need to be changed as well from receiver to exporter